### PR TITLE
deploy.sh made slightly faster (shorter downtime)

### PR DESCRIPTION
**Pull request by pwilczynskiclearcode**

### CODE REVIEW

File: {{cookiecutter.repostory_name}}/app/Dockerfile [hunk 1]
```diff
@@ -23,6 +23,7 @@ RUN pip3 install --user --no-cache-dir -r requirements.txt
 COPY ./src/ /root/src/
 
 RUN python3 -m compileall -b -f -q /root/
+RUN SECRET_KEY=0 CSP_ENABLED= DATABASE_URL= python3 manage.py collectstatic --no-input --clear
```
Comment (pwilczynskiclearcode) [inline {{cookiecutter.repostory_name}}/app/Dockerfile]: By tricking with these env variables we can run `collectstatic` during the container build

File: {{cookiecutter.repostory_name}}/dc-prod.yml [hunk 2]
```diff
@@ -36,8 +36,8 @@ services:
     restart: always
     env_file: ./.env
     volumes:
-      - backend-static:/srv/app/static
-      - ./media:/srv/app/media
+      - backend-static:/root/src/static
+      - ./media:/root/src/media
```
Comment (pwilczynskiclearcode) [inline {{cookiecutter.repostory_name}}/dc-prod.yml]: these path should have been changed before, my mistake

File: {{cookiecutter.repostory_name}}/deploy.sh [hunk 3]
```diff
@@ -2,10 +2,18 @@
 PRIMARY_IMAGE_HASH=$(docker-compose build app | tee /dev/tty \
     | sed '/^$/d' | grep "AS secondary-image" -B1 | head -1 | awk '{print $2}')
 docker-compose build  # build the rest of images
-docker-compose down -v --remove-orphans
```
Comment (pwilczynskiclearcode) [inline {{cookiecutter.repostory_name}}/deploy.sh]: `-v` is for stopping volumes. I don't think we really need to stop them when restarting the services

File: {{cookiecutter.repostory_name}}/deploy.sh [hunk 4]
```diff
@@ -2,10 +2,18 @@
...
+{% if cookiecutter.use_celery == "y" -%}
+    celery-worker celery-beat \
+{%- endif %}
+{% if cookiecutter.use_flower == "y" -%}
+    celery-flower
+{%- endif %}
+
 docker-compose exec app python manage.py migrate
-docker-compose exec app python manage.py collectstatic
+docker-compose exec nginx nginx -s reload
```
Comment (pwilczynskiclearcode) [inline {{cookiecutter.repostory_name}}/deploy.sh]: I doubt there are many of changes to the `nginx` service declaration in `docker-compose.yml` itself. I believe that most of the changes will be done to the nginx config so reloading it without the process downtime might be much better as it won't terminate connections established between clients and nginx

File: {{cookiecutter.repostory_name}}/dc-prod.yml [hunk 5]
```diff
@@ -11,7 +11,7 @@ services:
     logging: &logging
       driver: journald
       options:
-        tag: '{{.Name}}'
+        tag: {% raw -%}'{{.Name}}'{%- endraw %}
```
Comment (pwilczynskiclearcode) [inline {{cookiecutter.repostory_name}}/dc-prod.yml]: without the `{% raw %}` cookiecutter tries to find `.Name` variable in its configuration

**Comment by pwilczynskiclearcode** [comment]
@ppolewicz Could you have a look?

File: {{cookiecutter.repostory_name}}/deploy.sh (hunk 3 above)
Comment (ppolewicz) [reply {{cookiecutter.repostory_name}}/deploy.sh]: nope, lets remove the flag if you don't think we need it

File: {{cookiecutter.repostory_name}}/app/Dockerfile (hunk 1 above)
Comment (ppolewicz) [reply {{cookiecutter.repostory_name}}/app/Dockerfile]: sure, but please add it as a comment

File: {{cookiecutter.repostory_name}}/deploy.sh [hunk 6]
```diff
@@ -2,10 +2,26 @@
...
+
+docker-compose stop app \
+{% if cookiecutter.use_celery == "y" -%}
+    celery-worker celery-beat \
+{%- endif %}
+{% if cookiecutter.use_flower == "y" -%}
+    celery-flower
+{%- endif %}
+
+docker-compose start app \
```
Comment (ppolewicz) [inline {{cookiecutter.repostory_name}}/deploy.sh]: https://stackoverflow.com/a/33716320/95920 says it won't create the container in the first place. Can we use `up`?

File: {{cookiecutter.repostory_name}}/deploy.sh [hunk 7]
```diff
@@ -2,10 +2,26 @@
...
+{% if cookiecutter.use_celery == "y" -%}
+    celery-worker celery-beat \
+{%- endif %}
+{% if cookiecutter.use_flower == "y" -%}
+    celery-flower
+{%- endif %}
```
Comment (ppolewicz) [inline {{cookiecutter.repostory_name}}/deploy.sh]: is there some way in which we can avoid manually enumerating container names? skip the nginx and do everything else?

File: {{cookiecutter.repostory_name}}/deploy.sh (hunk 4 above)
Comment (ppolewicz) [reply {{cookiecutter.repostory_name}}/deploy.sh]: sure, but please add a comment

File: {{cookiecutter.repostory_name}}/deploy.sh (hunk 6 above)
Comment (pwilczynskiclearcode) [reply {{cookiecutter.repostory_name}}/deploy.sh]: Yes, actually I don't need to specify services when running `up`. Every single service needs to be up

File: {{cookiecutter.repostory_name}}/deploy.sh (hunk 7 above)
Comment (pwilczynskiclearcode) [reply {{cookiecutter.repostory_name}}/deploy.sh]: So I resigned from jinja variables in favour of listing services using `docker-compose ps`

**Comment by pwilczynskiclearcode** [comment]
@ppolewicz Could you look at the changes once again?

**Pull request merged**